### PR TITLE
feature(docker): run docker artifact test ontop of FIPS

### PR DESCRIPTION
**Pull request by fruch**
in this change we introduce a new AWS ASG for testing
docker artifacts on top of FIPS machines,
it has a separated SCT runner, and ASG and labels in jenkins.

so it would be completely separate from the regular SCT runner images,
and can be updated separately

### Testing

- [x] 🟢  docker with fips - https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/fruch/job/artifacts-docker-fips-test/12/
- [x] 🔴 docker with fips 2024.1, crashing (what this case is for): https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/fruch/job/artifacts-docker-fips-test/16/

### PR pre-checks (self review)

- [x] I added the relevant `backport` labels
- [x] I didn't leave commented-out/debugging code

### Reminders

- Add New configuration option and document them (in `sdcm/sct_config.py`)
- Add unit tests to cover my changes (under `unit-test/` folder)
- Update the Readme/doc folder relevant to this change (if needed)

### CODE REVIEW

**Comment by fruch** [comment]
this depends on #8290 to be merged first.

File: sdcm/sct_runner.py [hunk 1]
```diff
@@ -1349,3 +1352,8 @@ def clean_sct_runners(test_status: str,
             end_message = "No runners have been terminated"
 
     LOGGER.info(end_message)
+
+
+class AwsFipsSctRunner(AwsSctRunner):
+    VERSION = f"{SctRunner.VERSION}-fips"
+    BASE_IMAGE = 'resolve:ssm:/aws/service/marketplace/prod-k6fgbnayirmrc/latest'
```
Comment (roydahan) [inline sdcm/sct_runner.py:1359]: How do we know that when we start this image the fips is enabled?
Is there a way to check that in the test maybe?

Comment (fruch) [reply sdcm/sct_runner.py:1359]: We have that in the fips artifact test.

I need to check when it's done, I think if we identify it's Ubuntu Pro, when check it

Anyhow it's the official product from the Ubuntu title as fips, and so far we took a specific AMI out then, this syntax enables us to get the latest each time.

Anyhow we aren't gonna build those too often

Comment (fruch) [reply sdcm/sct_runner.py:1359]: One more "proof" for now,  that it reproduces the crash, that is being worked on. (Not sure by whom, that might would want to try it out as part of investigations and fixes)

Comment (fruch) [reply sdcm/sct_runner.py:1359]: here is the check
https://github.com/scylladb/scylla-cluster-tests/commit/64b4e506ff4a17e7bf9d6ffe9c04242b88162460

Comment (fruch) [reply sdcm/sct_runner.py:1359]: but when I think of it, it's relevant to the db nodes
no to the sct-runner or the builder

**Review by roydahan — APPROVED** [review]

**Comment by fruch** [comment]
@yaronkaikov 

FYI we know have a new test that can cover FIPS docker ontop of FIPS machine.
If you want to start triggering it on enterprise

Also backporting it to 2023.1/2024.1/2024.2 (job for it would be be need to be created manually)

**Comment by Annamikhlin** [comment]
@fruch - seems it wasn't backported to 2024.2

**Comment by fruch** [comment]
> @fruch - seems it wasn't backported to 2024.2

o.k. let's get it backported then

**Pull request merged**